src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,32 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static animal = [];
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.animal.push(this);
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (!(animal instanceof Carnivore) && animal.hidden === false) {
```

Comment:
The condition should check if the `animal` is an instance of `Herbivore`, not just that it is not an instance of `Carnivore`. According to the task requirements, it's enough to check that the incoming object inside the `bite` method is an instance of class `Herbivore`.